pallets/salp/src/lib.rs

```diff
@@ -1022,6 +1032,62 @@ pub mod pallet {
 
 			Ok(())
 		}
+
+		/// transfer asset to statemine
+		#[pallet::weight(T::WeightInfo::contribute())]
+		pub fn transfer_statemine_assets(
+			origin: OriginFor<T>,
+			amount: BalanceOf<T>,
+			asset_id: u32,
+		) -> DispatchResult {
+			let who = ensure_signed(origin)?;
+			let origin_location = T::AccountIdToMultiLocation::convert(who.clone());
+			let mut assets = MultiAssets::new(); // VersionedMultiAssets::V1(MultiAssets::new(MultiAsset))
+			let amount_128 =
+				TryInto::<u128>::try_into(amount).map_err(|_| Error::<T>::ConvertFailure)?;
+			let statemine_asset = MultiAsset {
+				id: AssetId::Concrete(MultiLocation::new(
+					1,
+					Junctions::X2(
+						Junction::Parachain(1000),
+						Junction::GeneralIndex(asset_id.into()),
+					),
+				)),
+				fun: Fungibility::Fungible(amount_128),
+			};
+			let dest_weight = 4 * T::BaseXcmWeight::get();
+			let fee_asset = MultiAsset {
+				id: AssetId::Concrete(MultiLocation::new(1, Junctions::Here)),
+				fun: Fungibility::Fungible(dest_weight.into()),
+			};
```

Comment:
I would be surprised if the weight is the correct amount here. If it is it's a coincidence. You'll want to check Statemine's weight to fee calculation [here](https://github.com/paritytech/cumulus/blob/master/polkadot-parachains/statemine/src/constants.rs#L57-L71). Though this might be a decent overestimate for now.

Reply:
yes,should use Statemine's `weightToFee` here and we may revamp later but before [4470](https://github.com/paritytech/polkadot/pull/4470) released we should just disable this call